src/file/tokio_compat_file.rs

```diff
@@ -474,22 +511,25 @@ impl AsyncWrite for TokioCompatFile {
             .unwrap_or(max_write_len);
 
         let write_limit = self.get_auxiliary().tokio_compat_file_write_limit();
-        let write_len = self.as_mut().project().write_len;
-        match write_len.checked_add(n as usize) {
-            Some(new_write_len) if new_write_len > write_limit => {
-                ready!(self.as_mut().poll_flush(cx))?;
-            }
-            None => {
-                // case overflow
-                // This has to be a separate cases since
-                // write_limit could be set to usize::MAX, in which case
-                // saturating_add would never return anything larger
-                // than it.
-                ready!(self.as_mut().poll_flush(cx))?;
-            }
-            Some(new_write_len) => {
-                *write_len = new_write_len;
+        let mut write_len = self.as_mut().write_len;
```

Comment:
We could write part of the buffer instead of the entire buffer

Reply:
It seems that all previous write buffers have already sent to flush task. What we can control is only when to flush the return message. Is there any way to make `flush_task` write part of the buffer?

Reply:
eeh what I meant is that, instead of flushing right away, we can take a slice  of the buffer given in poll_write and write it.